src/benchmarks/micro/libraries/Microsoft.Extensions.Caching.Memory/MemoryCacheTests.cs

```diff
@@ -39,5 +43,66 @@ public void Setup()
 
         [Benchmark]
         public object SetOverride() => _memCache.Set("512", "512");
+
+        [GlobalSetup(Targets = new[] { nameof(AddThenRemove_NoExpiration), nameof(AddThenRemove_AbsoluteExpiration), nameof(AddThenRemove_RelativeExpiration) })]
+        public void Setup_AddThenRemove()
+        {
+            _items = ValuesGenerator.ArrayOfUniqueValues<int>(100).Select(x => ((object)x.ToString(), (object)x.ToString())).ToArray();
+        }
+
+        [Benchmark]
+        public void AddThenRemove_NoExpiration()
+        {
+            using (MemoryCache cache = new MemoryCache(new MemoryCacheOptions()))
+            {
+                foreach (var item in _items)
+                {
+                    cache.Set(item.key, item.value);
+                }
+
+                foreach (var item in _items)
+                {
+                    cache.Remove(item.key);
+                }
+            }
+        }
+
+        [Benchmark]
+        public void AddThenRemove_AbsoluteExpiration()
+        {
+            DateTimeOffset absolute = DateTimeOffset.UtcNow.AddHours(1);
+
+            using (MemoryCache cache = new MemoryCache(new MemoryCacheOptions()))
+            {
+                foreach (var item in _items)
+                {
+                    cache.Set(item.key, item.value, absolute);
+                }
+
+                foreach (var item in _items)
+                {
+                    cache.Remove(item.key);
+                }
+            }
+        }
+
+        [Benchmark]
+        public void AddThenRemove_RelativeExpiration()
+        {
+            TimeSpan relative = TimeSpan.FromHours(1);
+
+            using (MemoryCache cache = new MemoryCache(new MemoryCacheOptions()))
+            {
+                foreach (var item in _items)
+                {
+                    cache.Set(item.key, item.value, relative);
+                }
+
+                foreach (var item in _items)
+                {
+                    cache.Remove(item.key);
+                }
+            }
+        }
```

Comment:
Should we add a `SlidingExpiration` test as well? The above test is for `absoluteExpirationRelativeToNow` AFAICT.

Reply:
@eerhardt we should, thanks for pointing this out! I could not find an [extension method](https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs) that would allow for setting `SlidingExpiration` so I've changed all the benchmarks to use the `using` pattern for adding the entries (used by the extension methods)